docs/development/architecture.md

```diff
@@ -0,0 +1,180 @@
+---
+title: Architecture
+sidebar_position: 0
+---
+
+
+## Introduction
+
+Headlamp is a user-friendly and extensible Kubernetes UI designed to provide a seamless experience for managing Kubernetes clusters. It can connect to one or multiple Kubernetes clusters and can be deployed as a web app in-cluster or as a desktop app. Headlamp can be customized and 
+
+## High-Level Architecture
+
+### Headlamp running as a Desktop App (on MacOS, Windows, Linux etc)
+```mermaid
+graph TD
+    A[User] -->|Interacts with| B[Headlamp App]
+    B -->|Runs| C[Frontend, React]
+    B -->|Runs| D[Backend headlamp-server, Go]
+    D -->|Proxies| E[Kubernetes API]
+    C -->|Loads| F[Plugins]
+    D -->|Serves static web files and provides services to| C
+```
+
+### Headlamp Web App running inside a Kubernetes Cluster
+
+```mermaid
+graph TD
+    A[User] -->|Interacts with| G[Ingress/Gateway API]
+    G -->|Routes to| B[Frontend, React]
+    B -->|Sends requests to| C[Backend headlamp-server, Go]
+    C -->|Proxies| D[Kubernetes API]
+    C -->|Interacts with| E[Cluster Services]
+    B -->|Loads| F[Plugins]
+    C -->|Serves static web files and provides services to| B
+    subgraph Kubernetes Cluster
+        B
+        C
+        E
+    end
+```
+
+### Headlamp and websockets
+
+Headlamp can use websockets to get soft-realtime UI updates from Kubernetes when things change.
+
+Because web browsers limit the number of websockets that can be created, the headlamp-server 
+
+```mermaid
+graph LR
+    A[Browser] <-->|WebSocket| B[headlamp_server]
+    B <-->|WebSocket| C[Kubernetes API Server]
+    B <-->|WebSocket| D[Kubernetes API Server]
+    B <-->|WebSocket| E[Kubernetes API Server]
+```
+
+### Plugins
+
+#### Plugin Hosting Options
+
+Where can plugins be hosted?
+
+```mermaid
+graph TD
+    F[Plugins] -->|Can be| I[Private and Self Hosted]
+    F -->|Or| J[Public and Hosted on Artifact Hub]
```

Comment:
Not really hosted but indexed. Hosted in Github, Gitlab, or BitBucket, and indexed in ArtifactHub.